src/main/java/it/gov/pagopa/payhub/activities/activity/ingestionflow/BaseIngestionFlowFileActivity.java

```diff
@@ -51,36 +50,36 @@ public T processFile(Long ingestionFlowFileId) {
 	}
 
 	/**
-	 * Retrieves the {@link IngestionFlowFileDTO} record for the given ID. If no record is found, throws
+	 * Retrieves the {@link IngestionFlowFile} record for the given ID. If no record is found, throws
 	 * an {@link IngestionFlowFileNotFoundException}. Validates the flow file type before returning.
 	 *
 	 * @param ingestionFlowFileId the ID of the ingestion flow file to retrieve
-	 * @return the {@link IngestionFlowFileDTO} corresponding to the given ID
+	 * @return the {@link IngestionFlowFile} corresponding to the given ID
 	 * @throws IngestionFlowFileNotFoundException if the record is not found
 	 * @throws IllegalArgumentException if the flow file type is invalid
 	 */
-	private IngestionFlowFileDTO findIngestionFlowFileRecord(Long ingestionFlowFileId) {
-		IngestionFlowFileDTO ingestionFlowFileDTO = ingestionFlowFileDao.findById(ingestionFlowFileId)
+	private IngestionFlowFile findIngestionFlowFileRecord(Long ingestionFlowFileId) {
+		IngestionFlowFile ingestionFlowFileDTO = ingestionFlowFileService.findById(ingestionFlowFileId)
 			.orElseThrow(() -> new IngestionFlowFileNotFoundException("Cannot found ingestionFlow having id: "+ ingestionFlowFileId));
 
-		if (!getHandledIngestionFlowFileType().equals(ingestionFlowFileDTO.getFlowFileType())) {
+		if (!(getHandledIngestionFlowFileType().name()).equals(ingestionFlowFileDTO.getFlowFileType().name())) {
```

Comment:
aren't these the same enum type?
```suggestion
		if (!(getHandledIngestionFlowFileType()).equals(ingestionFlowFileDTO.getFlowFileType())) {
```